dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/datasource/DatabasePoolUtils.java

```diff
@@ -252,6 +259,34 @@ private static DataSource createHikariDbPool(PoolConfig config) throws SQLExcept
     return ds;
   }
 
+  private static DataSource createUnpooledDataSource(PoolConfig config) throws SQLException {
+    ConfigKeyMapper mapper = config.getMapper();
+
+    DhisConfigurationProvider dhisConfig = config.getDhisConfig();
+
+    final String driverClassName =
+        dhisConfig.getProperty(mapper.getConfigKey(CONNECTION_DRIVER_CLASS));
+    final String jdbcUrl =
+        firstNonNull(
+            config.getJdbcUrl(), dhisConfig.getProperty(mapper.getConfigKey(CONNECTION_URL)));
+    final String username =
+        firstNonNull(
+            config.getUsername(), dhisConfig.getProperty(mapper.getConfigKey(CONNECTION_USERNAME)));
+    final String password =
+        firstNonNull(
+            config.getPassword(), dhisConfig.getProperty(mapper.getConfigKey(CONNECTION_PASSWORD)));
```

Comment:
There are many duplicated lines in this class, could we possibly reuse some code here?
Using sonarlint locally identifies these.